controllers/ibu_controller.go

```diff
@@ -99,40 +103,38 @@ func requeueWithCustomInterval(interval time.Duration) ctrl.Result {
 //
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.8.3/pkg/reconcile
-func (r *ImageBasedUpgradeReconciler) Reconcile(ctx context.Context, req ctrl.Request) (nextReconcile ctrl.Result, err error) {
+func (r *ImageBasedUpgradeReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	r.Log.Info("Start reconciling IBU", "name", req.NamespacedName)
-	defer func() {
-		if nextReconcile.RequeueAfter > 0 {
-			r.Log.Info("Finish reconciling IBU", "name", req.NamespacedName, "requeueAfter", nextReconcile.RequeueAfter.Seconds())
-		} else {
-			r.Log.Info("Finish reconciling IBU", "name", req.NamespacedName, "requeueRightAway", nextReconcile.Requeue)
-		}
-	}()
 
-	nextReconcile = doNotRequeue()
+	// default to not requeue
+	nextReconcile := doNotRequeue()
 
 	if req.Name != utils.IBUName {
-		return
+		msg := fmt.Errorf("invalid IBU name, it must be named '%s'", utils.IBUName)
+		return doNotRequeue(), msg
```

Comment:
If you return msg as an error, k8s will retry by itself.

Reply:
Updated to return nil